### PR TITLE
Fix checkbox

### PR DESCRIPTION
**Pull request by nearbycoder**
Fixes the issue where our app was causing problems with unknown props being passed to the checkbox. Additionally this speeds up our spec times.

### CODE REVIEW

File: lib/components/checkbox/checkbox.theme.js [hunk 1]
```diff
@@ -168,7 +168,9 @@ export const Checkbox = {
           color: 'red.70',
         },
       },
-      icon: {},
+      icon: {
+        color: 'grey.0',
```
Comment (nearbycoder) [inline lib/components/checkbox/checkbox.theme.js:172]: the icon has always been white but I noticed that the latest storybook had it black, this was to fix that.

File: lib/components/icons/Icon.jsx [hunk 2]
```diff
@@ -125,7 +125,8 @@ export const iconList = [
   'warning',
 ];
 
-export const Icon = ({ name, ...props }) => {
+// eslint-disable-next-line no-unused-vars
+export const Icon = ({ name, isIndeterminate, isChecked, ...props }) => {
```
Comment (hizegi) [inline lib/components/icons/Icon.jsx:129]: thank you.

**Review by hizegi — APPROVED** [review]

**Review by theJakub — APPROVED** [review]
Good catch here. I wonder if this should start a conversation about not using `...props`. Forcing us to be more specific about what we are passing. I know that makes DS less flexible without updates, but may protect against these types of issues.

**Comment by nearbycoder** [comment]
@theJakub that is a great discussion! I think we are off next Monday but for sure need to add it to the guild discussion!

**Pull request merged**